scene/animation/animation_blend_tree.cpp

```diff
@@ -247,6 +249,7 @@ void AnimationNodeOneShot::get_parameter_list(List<PropertyInfo> *r_list) const
 	r_list->push_back(PropertyInfo(Variant::FLOAT, remaining, PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NONE));
 	r_list->push_back(PropertyInfo(Variant::FLOAT, fade_out_remaining, PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NONE));
 	r_list->push_back(PropertyInfo(Variant::FLOAT, time_to_restart, PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NONE));
+	APPEND_FILTER_PARAMETER(r_list);
```

Comment:
Why do some animation nodes have a filter and some don't?

I find this use of macro really confusing. Why can't we have the "filter" parameter in the base class `AnimationNode`?

If we do have a special type of "filterable" animationnode, perhaps we should add a virtual class / level of inheritance to indicate the filterability (having the "filter" parameter).

Reply:
Because I "filter" just a normal parameter like "add_amount" in `AnimationNodeAdd2`, I think it should be added where they need in `get_parameter_list()`, this macros just for copying duplicate code and avoiding stupid mistake.

However, if user create their own `AnimationNode` in GDScript ([godot/pull/69641](https://github.com/godotengine/godot/pull/69641)) and need filter, it may not work expectantly because of misspell.
For user friendly, I readd `has_filter()` now.

But I am not sure which way is more reasonable.